### PR TITLE
Make cluster-monitoring operator priority class aware.

### PR DESCRIPTION
**Pull request by aveshagarwal**
@sjenning 

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1624253

### CODE REVIEW

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -372,6 +377,10 @@ func (f *Factory) KubeStateMetricsDeployment() (*appsv1.Deployment, error) {
 		d.Spec.Template.Spec.NodeSelector = f.config.KubeStateMetricsConfig.NodeSelector
 	}
 
+	if len(f.config.KubeStateMetricsConfig.PriorityClassName) > 0 {
```
Comment (squat) [inline pkg/manifests/manifests.go]: for consistency, could we change all of these string comparisons from `len(x) > 0` to `x != ""`?

**Comment by brancz** [comment]
I don't think this should be user configurable, these are system pods and should just always be the cluster-critical priority class.

**Comment by brancz** [comment]
(also sorry, my bad for not seeing this PR and just opening #99)

**Comment by aveshagarwal** [comment]
> I don't think this should be user configurable, 

If you mean end user, I agree. But IMO, cluster admin should be able to configure what priority they want for their clusters. Also in 3.11, we need a way to configure these priority classes for cluster-monitoring-operator stack via openshift-ansible at the time of installation, for example here: https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_cluster_monitoring_operator/templates/cluster-monitoring-operator-config.j2 ? 
 
> these are system pods and should just always be the cluster-critical priority class.

Please note system-cluster-critical is not the highest priority, system-node-critical is the highest. In general, we have assigned system-node-critical to only those pods (like sdn, etcd, apiserver, etc..) and other critical pods gets system-cluster-critical.

**Comment by aveshagarwal** [comment]
> I don't see why it needs to be configurable, 

A cluster admin might have to decide between system-cluster-critical and system-node-critical depending upon their requirements. But I am fine now if we dont want to make it configurable, as far as we assign a priority class by default. Are we doing it?

**Comment by brancz** [comment]
Sadly as I said for the ones that would actually be important (Prometheus/Alertmanager) it's a lot more complicated to get this down as we would have to do a minor version release and introduce all of these new versions before tomorrow evening, which is when code freeze is. So we're moving this to z-stream of 3.11.

**Comment by paulfantom** [comment]
Closing as this went stale a long time ago. Please reopen (or create JIRA ticket) if still needed.

/close

**Pull request closed without merging**